bundle/config/mutator/process_target_mode.go

```diff
@@ -30,96 +26,66 @@ func (m *processTargetMode) Name() string {
-			r.Jobs[i].Schedule.PauseStatus = jobs.PauseStatusPaused
-		}
-		if r.Jobs[i].Continuous != nil && r.Jobs[i].Continuous.PauseStatus != jobs.PauseStatusUnpaused {
-			r.Jobs[i].Continuous.PauseStatus = jobs.PauseStatusPaused
-		}
-		if r.Jobs[i].Trigger != nil && r.Jobs[i].Trigger.PauseStatus != jobs.PauseStatusUnpaused {
-			r.Jobs[i].Trigger.PauseStatus = jobs.PauseStatusPaused
-		}
-	}
 
-	for i := range r.Pipelines {
-		r.Pipelines[i].Name = prefix + r.Pipelines[i].Name
-		r.Pipelines[i].Development = true
-		// (pipelines don't yet support tags)
+	if t.Prefix == "" {
+		b.Config.Transform.Prefix = "[dev " + shortName + "] "
 	}
 
-	for i := range r.Models {
-		r.Models[i].Name = prefix + r.Models[i].Name
-		r.Models[i].Tags = append(r.Models[i].Tags, ml.ModelTag{Key: "dev", Value: tagValue})
+	if t.Tags == nil {
+		b.Config.Transform.Tags = &map[string]string{
+			"dev": b.Tagging.NormalizeValue(shortName),
+		}
 	}
 
-	for i := range r.Experiments {
-		filepath := r.Experiments[i].Name
-		dir := path.Dir(filepath)
-		base := path.Base(filepath)
-		if dir == "." {
-			r.Experiments[i].Name = prefix + base
-		} else {
-			r.Experiments[i].Name = dir + "/" + prefix + base
-		}
-		r.Experiments[i].Tags = append(r.Experiments[i].Tags, ml.ExperimentTag{Key: "dev", Value: tagValue})
+	if t.DefaultJobsMaxConcurrentRuns == 0 {
+		t.DefaultJobsMaxConcurrentRuns = developmentConcurrentRuns
 	}
 
-	for i := range r.ModelServingEndpoints {
-		prefix = "dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"
-		r.ModelServingEndpoints[i].Name = prefix + r.ModelServingEndpoints[i].Name
-		// (model serving doesn't yet support tags)
+	if t.DefaultTriggerPauseStatus == "" {
+		t.DefaultTriggerPauseStatus = config.Paused
 	}
 
-	for i := range r.RegisteredModels {
-		prefix = "dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"
-		r.RegisteredModels[i].Name = prefix + r.RegisteredModels[i].Name
-		// (registered models in Unity Catalog don't yet support tags)
+	if !config.IsExplicitlyDisabled(t.DefaultPipelinesDevelopment) {
+		enabled := true
+		t.DefaultPipelinesDevelopment = &enabled
 	}
 
 	return nil
 }
 
-func disableDeploymentLock(b *bundle.Bundle) error {
-	return b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
-		return dyn.Map(v, "bundle.deployment.lock", func(_ dyn.Path, v dyn.Value) (dyn.Value, error) {
-			return dyn.Set(v, "enabled", dyn.V(false))
-		})
-	})
-}
-
 func validateDevelopmentMode(b *bundle.Bundle) diag.Diagnostics {
+	t := b.Config.Transform
+	u := b.Config.Workspace.CurrentUser
+
+	// Make sure everything is paused by default to avoid surprises
+	if t.DefaultTriggerPauseStatus == config.Unpaused {
+		return diag.Diagnostics{{
+			Severity: diag.Error,
+			Summary:  "target with 'mode: development' cannot set trigger pause status to UNPAUSED by default",
+			Location: b.Config.GetLocation("transform.default_trigger_pause_status"),
+		}}
+	}
```

Comment:
Why? The other settings can be overridden.

Reply:
It's not very different and I'm not 100% sure there was sufficient evidence for us to allow that. But that behavior exists today so all we can do is add a warning or add a special setting to ignore those overrides for certain deployments. The warning doesn't really seem warranted since users are so unlikely to set that field by mistake. But I think we may want that extra setting/command line arg at some point.

Reply:
Fair enough. I suspect this will be found surprising by >= 1 because of the asymmetry, but because it is local to the development mode we can run with it.